packages/relay/src/lib/clients/sdkClient.ts

```diff
@@ -135,9 +135,18 @@ export class SDKClient {
             .setAccountId(AccountId.fromString(address)), this.clientMain, callerName, address, requestId);
     }
 
-    async getContractByteCode(shard: number | Long, realm: number | Long, address: string, callerName: string, requestId?: string): Promise<Uint8Array> {
-        return this.executeQuery(new ContractByteCodeQuery()
-            .setContractId(ContractId.fromEvmAddress(shard, realm, address)), this.clientMain, callerName, address, requestId);
+    async getContractByteCode(shard: number | Long, realm: number | Long, address: string, callerName: string, hasProhibitedOpCodes: boolean, requestId?: string): Promise<Uint8Array> {
+        if(hasProhibitedOpCodes) {
```

Comment:
Q: why the difference based on `hasProhibitedOpCodes`?
Shouldn't we query for cost regardless?

It should be possible to have a cost over the default even for prohibited contents no?

Reply:
It should be possible, but it is not, when I was debugging I found that the cost is calculated, but when doing `executeQuery` the behaviour of the sdk is different, my guess is that the sdk under the hood is doing something different when cost is provided.

Reply:
What's the difference?
What does the SDK return in the case of executeQuery when `hasProhibitedOpCOdes` is true?

Also maybe `hasProhibitedOpCOdes` is indicative of something else.
It also seems like we're assuming a precCheck error will occur if there are prohibitedOpCOdes, but I don't think that's assured. I think it's only applicable if `SELFDESTRUCT` has been executed on a contract but it may not have yet.

Reply:
when we do `executeQuery` with cost it throws errors

```
transaction 0.0.2@1689870961.095051289 failed precheck with status INVALID_CONTRACT_ID

transaction 0.0.2@1689871011.865656621 failed precheck with status CONTRACT_DELETED
```

As oppossed when we it has prohibited OpCodes but we don't define the cost, it returns the expected `ByteCode`

I did the path based on opCodes since that is where the difference lays, but maybe we should change the tests, and instead of expecting the ByteCodes, we should expect `0x` ?

Reply:
That's the challenge we can't make assumptions of the bytecode for some contracts so we need to make the call and see what we get.
A contract may have the prohibited opCodes but it may have not been executed and could be a valid contract with valid non 0x bytecode

Reply:
The the prohibited opCodes were only ever to dictate whether we could cache the bytecode or not because you might have valid bytecode at time T1 but after a SELFDESTRUCT is executed it would now be 0x on the mirror node or one of those precheck errors on the consensus node.

This flow is only happening because some `SELFDESTRUCT` cases were not migrated to mirror node so we could get valid bytecode back from the mirror node but the contract was deleted hence the call to the consensus node. In the backup flow we can't make any assumptions of what it could be

Reply:
Okay, I found the bug and removed the conditional path, good thing that we were able to find it 👍 
Thank you!